components/contract-components/tables/cells.tsx

```diff
@@ -1,52 +1,51 @@
-import {
-  ContractType,
-  PrebuiltContractType,
-  SchemaForPrebuiltContractType,
-} from "@thirdweb-dev/sdk";
-
 import { Skeleton } from "@chakra-ui/react";
-
-import React, { memo } from "react";
-
+import { memo } from "react";
 import { ChakraNextLink, Text } from "tw-components";
-
-import { useContractMetadataWithAddress } from "@3rdweb-sdk/react";
-
 import { useChainSlug } from "hooks/chains/chainSlug";
-
-import { z } from "zod";
 import { shortenIfAddress } from "utils/usedapp-external";
 import { usePublishedContractsFromDeploy } from "../hooks";
+import { useReadContract } from "thirdweb/react";
+import { getContractMetadata, name } from "thirdweb/extensions/common";
+import { getContract, defineChain } from "thirdweb";
+import { thirdwebClient } from "lib/thirdweb-client";
+import { BasicContract } from "contract-ui/types/types";
 
 interface AsyncContractNameCellProps {
-  cell: {
-    address: string;
-    chainId: number;
-    metadata: () => Promise<
-      z.infer<SchemaForPrebuiltContractType<PrebuiltContractType>["output"]>
-    >;
-  };
+  cell: BasicContract;
 }
 
 export const AsyncContractNameCell = memo(
   ({ cell }: AsyncContractNameCellProps) => {
     const chainSlug = useChainSlug(cell.chainId);
-    const metadataQuery = useContractMetadataWithAddress(
-      cell.address,
-      cell.metadata,
-      cell.chainId,
-    );
+
+    const contract = getContract({
+      client: thirdwebClient,
+      address: cell.address,
+      chain: defineChain(cell.chainId),
+    });
+
+    const contractMetadata = useReadContract(getContractMetadata, {
+      contract,
+    });
+
+    const contractName = useReadContract(name, {
+      contract,
+    });
```

Comment:
abstract this into a custom hook, you can probably make ur own `useQuery()` hook where u call both getContractMetadata and name extensions in parallel and merge them